javascript/new-src/src/js/app.js

```diff
@@ -14,665 +14,698 @@
+              .then(function(response) {
+                return response.data;
+              });
+          },
+          // /json/rdm/uid_info?id=[universe]&uid=[uid]
+          UidInfo: function(universe, uid) {
+            var url = '/json/rdm/uid_info' +
+              '?id=' + universe + '&uid=' + uid;
+            return $http.get(url)
+              .then(function(response) {
+                return response.data;
+              });
+          },
+          // /json/rdm/uid_personalities?id=[universe]&uid=[uid]
+          UidPersonalities: function(universe, uid) {
+            var url = '/json/rdm/uid_personalities' +
+              '?id=' + universe + '&uid=' + uid;
+            return $http.get(url)
+              .then(function(response) {
+                return response.data;
+              });
+          },
+          // /json/rdm/uids?id=[universe]
+          Uids: function(universe) {
+            var url = '/json/rdm/uids?id=' + universe;
+            return $http.get(url)
+              .then(function(response) {
+                return response.data;
+              });
+          },
+          // /rdm/run_discovery?id=[universe]&incremental=true
+          RunDiscovery: function(universe, incremental) {
+            var url = '/rdm/run_discovery?id=' + universe;
+            if (incremental === true) {
+              url = url + '&incremental=true';
+            }
+            return $http.get(url).then(function(response) {
+              return response.data;
+            });
+          }
+        },
+        tabs: function(tab, id) {
+          // TODO(Dave_o): use a directive instead of this
+          $window.$('ul#ola-nav-tabs').html('' +
+            '<li id="overview" role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/\">Overview</a></li>' +
+            '<li id="faders" role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/faders">Faders</a></li>' +
+            '<li id="keypad" role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/keypad">Keypad</a></li>' +
+            '<li id="rdm" role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/rdm">RDM</a></li>' +
+            '<li id="patch" role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/patch">RDM Patcher</a></li>' +
+            '<li id="settings"role="presentation"><a href="/new/#/universe/' +
+            id +
+            '/settings">Settings</a></li>');
+          $window.$('ul#ola-nav-tabs > li#' + tab).addClass('active');
+        },
+        header: function(name, id) {
+          $('div#header-universe').html(
+            '<h4>' + name + '</h4><div>(id: ' + id + ')</div>'
+          );  // TODO(Dave_o): use a directive instead of this
+        },
+        error: {
+          modal: function(body, title) {
+            if (typeof body !== 'undefined') {
+              $('#errorModalBody').text(body);
```

Comment:
I plan or replacing this anyway with [this](https://angular-ui.github.io/bootstrap/#/modal)